components/org.wso2.carbon.config.configuration/src/test/java/org/wso2/carbon/config/configuration/configprovider/ConfigProviderImplTest.java

```diff
@@ -470,7 +441,36 @@ public void yamlConfigWithoutEnvValueTestCase() throws ConfigurationException {
         ConfigFileReader fileReader =
                 new YAMLBasedConfigFileReader(getFilePath("envconfigwithoutdefaults.yaml"));
         ConfigProvider configProvider = new ConfigProviderImpl(fileReader);
-        Configurations configurations = configProvider.getConfigurationObject(Configurations.class);
+        TestConfiguration configurations = configProvider.getConfigurationObject(TestConfiguration.class);
         Assert.assertNull(configurations, "configurations object should be null");
     }
+
+    /**
+     * Get file from resources.
+     *
+     * @param fileName name of the file
+     * @return file path
+     */
+    private Path getFilePath(String fileName) {
+        URL resourceURL = this.getClass().getClassLoader().getResource("conf");
+        if (resourceURL == null) {
+            throw new RuntimeException("Resource path not found");
+        }
+        return Paths.get(resourceURL.getPath(), fileName);
+    }
+
+    /**
+     * Set environmental variables.
+     */
+    private void setUpEnvironment() {
+        // This is how to set Environment Variables
```

Comment:
This comment might not be needed as it has already been explained in the method comment.